### ISSUE TITLE
investigate excessive memory usage

### ISSUE THREAD

**Issue by balhoff** [INCATools/dosdp-tools#274]
The Mondo makefile splits its pattern list into 3 groups; otherwise the memory growth is excessive. 

https://github.com/monarch-initiative/mondo/blob/39033f674d112c8d862d9b34f55d8803c8d15b2f/src/ontology/mondo.Makefile#L1-L26

This may relate to #273

cc @matentzn

**Comment by matentzn** [comment]
Thank you for looking into this!

**Comment by jdr0887** [comment]
I changed the mondo.Makefile to look like the following:

```
#ALL_PATTERNS_1=$(patsubst ../patterns/dosdp-patterns/%.yaml,%,$(wildcard ../patterns/dosdp-patterns/[a-h]*.yaml))                                                                                                                                            
#ALL_PATTERNS_2=$(patsubst ../patterns/dosdp-patterns/%.yaml,%,$(wildcard ../patterns/dosdp-patterns/[i-p]*.yaml))                                                                                                                                            
#ALL_PATTERNS_3=$(patsubst ../patterns/dosdp-patterns/%.yaml,%,$(wildcard ../patterns/dosdp-patterns/[q-z]*.yaml))                                                                                                                                            
ALL_PATTERNS=$(patsubst ../patterns/dosdp-patterns/%.yaml,%,$(wildcard ../patterns/dosdp-patterns/*.yaml))                                                                                                                                                    
#DOSDPT=../dosdp-tools-0.15.1-SNAPSHOT/bin/dosdp-tools                                                                                                                                                                                                        
DOSDPT=$(HOME)/workspace/github/INCATools/dosdp-tools-zio/target/universal/stage/bin/dosdp-tools                                                                                                                                                              
```
...snip...
```
#matches: matches_1 matches_2 matches_3                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                              
matches:                                                                                                                                                                                                                                                      
        $(DOSDPT) query --ontology=$(SRC) --catalog=catalog-v001.xml --reasoner=elk --obo-prefixes=true --batch-patterns="$(ALL_PATTERNS)" --template="../patterns/dosdp-patterns" --outfile="../patterns/data/matches/" 
```

I successfully ran this target with the latest dosdp-tools in about 1.5 hours & 30GB of memory.  I had tested first with 90GB, and then 45GB, and finally 30GB.  I assumed that most machines would/could have 30GB & no further testing at reduced memory was necessary.

**Comment by matentzn** [comment]
The tears of happiness this brings into my eyes! Thank you _very_ much!!

**Comment by balhoff** [comment]
Fixed by #279.
